src/pages/study-room/hooks/useExitStudyRoom.ts

```diff
@@ -11,15 +12,23 @@ interface UseExitStudyRoom {
 
 const useExitStudyRoom = ({ studyId, close }: UseExitStudyRoom) => {
   const navigate = useNavigate();
-  const { state: blockState, proceed } = useBlocker(
-    ({ currentLocation, nextLocation }) => currentLocation.pathname !== nextLocation.pathname,
-  );
 
-  const { mutate: exitStudyRoomMutate } = useExitStudyRoomMutation({ proceed, close, navigate });
+  useBlocker(({ currentLocation, nextLocation, historyAction }) => {
+    if (historyAction === 'POP') {
+      return currentLocation.pathname !== nextLocation.pathname;
+    }
+    return currentLocation.pathname === nextLocation.pathname;
+  });
+
+  const exitStudyRoomAction = () => {
+    close();
+    removeLocalStorage('participantId');
+    navigate(ROUTE_PATH.STUDY_ROOMS);
+  };
 
-  const exitStudyRoom = () => {
-    if (blockState !== 'blocked') return;
+  const { mutate: exitStudyRoomMutate } = useExitStudyRoomMutation({ exitStudyRoomAction });
```

Comment:
이렇게 표현되어야 내부 동작을 추측할 수 있을 것 같아요
```suggestion
  const { mutate: exitStudyRoomMutate } = useExitStudyRoomMutation({ onSuccess: exitStudyRoomAction });
```